nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/test/java/org/apache/nifi/processors/media/TestExtractMediaMetadata.java

```diff
@@ -0,0 +1,450 @@
+                fileAttrCount0++;
+            }
+        }
+        assertTrue(fileAttrCount0 > 1);
+
+        runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.MAX_NUMBER_OF_ATTRIBUTES, Integer.toString(fileAttrCount0 - 1));
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "txt.");
+        runner.assertValid();
+
+        runner.enqueue(new File("target/test-classes/textFile.txt").toPath());
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 1);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+
+        List<MockFlowFile> successFiles = runner.getFlowFilesForRelationship(ExtractMediaMetadata.SUCCESS);
+        MockFlowFile flowFile1 = successFiles.get(0);
+        int fileAttrCount1 = 0;
+        for (Map.Entry attr : flowFile1.getAttributes().entrySet()) {
+            if (attr.getKey().toString().startsWith("txt.")) {
+                fileAttrCount1++;
+            }
+        }
+        assertEquals(fileAttrCount0, fileAttrCount1 + 1);
+    }
+
+    @Test
+    public void testBmp() throws IOException {
+        final TestRunner runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "bmp.");
+        runner.assertValid();
+
+        runner.enqueue(new File("target/test-classes/16color-10x10.bmp").toPath());
+        runner.run(2);
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 1);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+
+        final List<MockFlowFile> successFiles = runner.getFlowFilesForRelationship(ExtractMediaMetadata.SUCCESS);
+        MockFlowFile flowFile0 = successFiles.get(0);
+        flowFile0.assertAttributeExists("filename");
+        flowFile0.assertAttributeEquals("filename", "16color-10x10.bmp");
+        flowFile0.assertAttributeExists("bmp.Content-Type");
+        flowFile0.assertAttributeEquals("bmp.Content-Type", "image/x-ms-bmp");
+        flowFile0.assertAttributeExists("bmp.X-Parsed-By");
+        assertTrue(flowFile0.getAttribute("bmp.X-Parsed-By").contains("org.apache.tika.parser.DefaultParser"));
+        assertTrue(flowFile0.getAttribute("bmp.X-Parsed-By").contains("org.apache.tika.parser.image.ImageParser"));
+        flowFile0.assertAttributeExists("bmp.height");
+        flowFile0.assertAttributeEquals("bmp.height", "10");
+        flowFile0.assertAttributeExists("bmp.width");
+        flowFile0.assertAttributeEquals("bmp.width", "10");
+    }
+
+    @Test
+    public void testJpg() throws IOException {
+        final TestRunner runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "jpg.");
+        runner.assertValid();
+
+        runner.enqueue(new File("target/test-classes/simple.jpg").toPath());
+        runner.run(2);
```

Comment:
Nope there shouldn't be (that's what the flowfile==null check at the start of the ontrigger is for), just wanted to be sure there wasn't supposed to be two files processed here or something. A comment explaining why it is run twice would be best.